docs/enhancements/scale-mode.md

```diff
@@ -0,0 +1,206 @@
+    reason: DeploymentFailure
+    status: "False"
+    type: Ready
+  targetNamespaces:
+  - lastTransitionTime: "2022-08-17T18:19:52Z"
+    message: Data plane kubeconfig secret data-plane-kubeconfig is missing
+    name: data-plane-002
+    reason: KubeconfigSecretMissing
+    result: Failure
+  - lastTransitionTime: "2022-08-17T18:19:52Z"
+    message: Namespace deployed successfully
+    name: data-plane-001
+    reason: SuccessfulDeployment
+    result: Success
+  - lastTransitionTime: "2022-08-17T18:19:52Z"
+    message: Namespace deployed successfully
+    name: data-plane-003
+    reason: SuccessfulDeployment
+    result: Success
+```
+
+## Namespaces
+The design decisions below are driven by the [OpenShift's documented maximums](https://docs.openshift.com/container-platform/4.10/scalability_and_performance/planning-your-environment-according-to-object-maximums.html); primarily the limit of 10,000 total namespaces per cluster.
+
+### On The Hub
+It is assumed that additional namespaces will be used by the consumer to house their operator(s), UI, etc.
+Enumerating/describing those is outside the scope of this document.
+
+#### Hive Operator
+The hive-operator manages the individual deployments.
+It can be deployed to a namespace in the control plane cluster as normal.
+
+#### Hive Deployments
+In scale mode, the hive-operator will look for namespaces with the following label:
+
+`hive.openshift.io/data-plane=true`
+
+Upon finding such a namespace, hive-operator will do the following:
+- Add the `hive.openshift.io/target-namespace=true` label.
+  This allows the operator to discover namespaces it should clean up when the `data-plane` label is removed.
+- Look for a Secret called `data-plane-kubeconfig`.
+  If not found, hive-operator will consider the namespace broken and not deploy the controller stack; and the HiveConfig `status.targetNamespaces` entry for this namespace will indicate failure.
+- Deploy the controller stack -- hive-controllers, hiveadmission, hive-clustersync -- to the namespace with the `data-plane-kubeconfig` secret mounted as a Volume in each pod.
+
+Summary of the **consumer's responsibilities**:
+- Create the namespace.
+- Stand up the data plane.
+  (Note that hive doesn't care where the data plane actually lives.
+  In particular, it may or may not be a service running in this namespace.)
+- Create the `data-plane-kubeconfig` Secret in the namespace.
+- Label the namespace with `hive.openshift.io/data-plane=true`
+
+What with eventual consistency, it's not crucial that the above be done in order; but adding the label last will result in the smoothest transition.
+Otherwise HiveConfig may temporarily report `Failure` status for the namespace e.g. if the Secret doesn't exist yet.
+
+As noted above, the consumer can ask hive to tear down the controllers for a data plane by removing the `data-plane` label from that namespace.
```

Comment:
Are data planes removed from `Status` when they are deleted/de-labeled?

Reply:
They probably are as a function of the tear down.

Reply:
Yes they are. The only way you would see them in hiveconfig status after they're unlabeled is if something goes wrong as we try to clean them up.